fixed-price-subscriptions/server/ruby/server.rb

```diff
@@ -49,39 +65,43 @@
   # Extract the price ID from environment variables given the name
   # of the price passed from the front end.
   #
-  # `price_id` should be an ID of a Price object on your account.
-  # In practice, you can also set a Price's `lookup_key` using
-  # the API when you create a Price, then fetch the list of prices
-  # by ID like so:
-  #
-  #   price_id = Stripe::Price.list(lookup_keys: ['basic']).data.first.id
-  price_id = ENV[data['priceLookupKey'].upcase]
-
-  begin
-    # Attach the payment method to the customer related
-    # to the authenticated user.
-    payment_method = Stripe::PaymentMethod.attach(
-      data['paymentMethodId'],
-      customer: customer_id
-    )
-  rescue Stripe::CardError => e
-    halt 400, { 'Content-Type' => 'application/json' }, { error: { message: e.error.message }}.to_json
-  end
+  # `price_id` is the an ID of a Price object on your account.
+  # This was populated using Price's `lookup_key` in the /config endpoint
+  price_id = data['priceId']
 
   # Create the subscription. Note we're using
   # expand here so that the API will return the Subscription's related
-  # latest invoice, and that latest invoice's payment_intent so that
-  # if SCA is required we can confirm the payment on the front end.
+  # latest invoice, and that latest invoice's payment_intent
+  # so we can collect payment information and confirm the payment on the front end.
   subscription = Stripe::Subscription.create(
-    default_payment_method: payment_method.id,
     customer: customer_id,
     items: [{
       price: price_id,
     }],
+    payment_behavior: 'default_incomplete',
     expand: ['latest_invoice.payment_intent']
   )
 
-  { subscription: subscription }.to_json
+  { subscriptionId: subscription.id, clientSecret: subscription.latest_invoice.payment_intent.client_secret }.to_json
+end
+
+post '/complete-subscription' do
```

Comment:
I added this as an illustration of setting the default payment method on the subscription. But this is really better handled via webhook because it's possible for a browser to close or request to have a network failure after confirming the payment intent but before completing this request.

What do you think? Leave this here or move it to a webhook and simplify `subscribe.js` even more?

Reply:
Decided to move to webhook